### PR TITLE
Remove checkpoint hash digests

### PR DESCRIPTION
**Pull request by evanlinjin**
Related to #32

### CODE REVIEW

File: bdk_core/tests/test_sparse_chain.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ fn checkpoint_limit_is_applied() {
     }
 
     assert_eq!(chain.iter_tx().count(), 10);
-    assert_eq!(chain.iter_checkpoints(..).count(), 5);
+    assert_eq!(chain.iter_checkpoints(..).count(), 10);
```
Comment (LLFourn) [inline bdk_core/tests/test_sparse_chain.rs:152]: I would actually try and preserve the checkpoint limit here unless it really gets in the way.

File: bdk_core/src/spk_tracker.rs [hunk 2]
```diff
@@ -52,31 +49,26 @@ impl<I: Clone + Ord> SpkTracker<I> {
...
-            .take_while(|checkpoint_id| {
-                !self
-                    .txid_digests_seen
-                    .contains(&chain.txid_digest_at(*checkpoint_id))
-            })
+            // .take_while(|checkpoint_id| {
+            //     !self
+            //         .txid_digests_seen
+            //         .contains(&chain.txid_digest_at(*checkpoint_id))
+            // })
```
Comment (LLFourn) [inline bdk_core/src/spk_tracker.rs:56]: delete rather than comment out this.

File: bdk_core/src/sparse_chain.rs [hunk 3]
```diff
@@ -322,13 +275,6 @@ impl SparseChain {
                 deepest_change = Some(deepest_change.unwrap_or(u32::MAX).min(change));
             }
         }
-        if let Some(change_depth) = deepest_change {
```
Comment (LLFourn) [inline bdk_core/src/sparse_chain.rs:325]: Delete deepest_change variable that is unused now above.

**Review by LLFourn — APPROVED** [review]
LGTM.

1. I think we could keep the checkpoint limit (it will be there eventually).
2. Preferable to just delete stuff rather than comment out for review.

Merge when ready.

**Comment by evanlinjin** [comment]
Probably will be replaced by #42

**Comment by evanlinjin** [comment]
Definitely will be replaced by #42

**Pull request closed without merging**